internal/vault/client.go

```diff
@@ -129,11 +129,10 @@ func NewClientFromStorageEntry(ctx context.Context, client ctrlclient.Client, en
 		return nil, fmt.Errorf("restored client's cacheKey %s does not match expected %s", cacheKey, entry.CacheKey)
 	}
 
-	if err := c.Validate(); err != nil {
-		return nil, err
-	}
+	c.Taint()
+	defer c.Untaint()
```

Comment:
I think with `defer c.Untaint()`, we always `Untaint()` either the client is valid or not. 
Should we `c.Untaint()` only when `c.Validate(ctx)` returns nil or before L139?

Reply:
Yeah, that was the prior approach. With the this change there should be no need to always untaint the client, since it was removed from the cache here: https://github.com/hashicorp/vault-secrets-operator/blob/a4d4f9f8f4397e9975eb5d340353e2ea404bccdf/internal/vault/client_factory.go#L443-L444